nbextensions/appCell2/widgets/appCellWidget.js

```diff
@@ -1235,48 +1258,31 @@ define(
                                 status: 'created',
                                 created: 0,
                             });
-                            newExecMessage = 'Launching...';
                             startListeningForJobMessages(message.job_id);
                             return fsmState.PROCESSING_LAUNCHED;
                         case 'error':
+                            model.setItem('appError', {
+                                type: 'App Startup Error',
+                                message: message.error_message,
+                                stacktrace: message.error_stacktrace,
+                                code: message.error_code,
+                                source: message.error_source,
+                                method: message.error_method,
+                                exceptionType: message.error_type,
+                            });
                             return fsmState.LAUNCH_ERROR;
                         default:
-                            throw new Error('Invalid launch state ' + message.event);
-                    }
-                })();
-                fsm.newState(newFsmState);
-                ui.setContent('run-control-panel.execMessage', newExecMessage);
-                renderUI();
-            }
-
-            function updateFromJobState(jobState, forceRender) {
-                const newFsmState = (function () {
-                    switch (jobState.status) {
-                        case 'created':
-                        case 'estimating':
-                        case 'queued':
-                            return fsmState.PROCESSING_QUEUED;
-                        case 'running':
-                            return fsmState.PROCESSING_RUNNING;
-                        case 'completed':
-                            stopListeningForJobMessages();
-                            return fsmState.COMPLETED;
-                        case 'terminated':
-                            stopListeningForJobMessages();
-                            return fsmState.TERMINATED;
-                        case 'error':
-                        case 'does_not_exist':
-                            stopListeningForJobMessages();
-                            return fsmState.RUNTIME_ERROR;
-                        default:
-                            throw new Error('Invalid job state ' + jobState.status);
+                            console.warn(`Unknown run-status event ${message.event}!`, message);
+                            model.setItem('appError', {
+                                type: 'App Startup Error',
+                                message: 'App launch returned an invalid response',
+                                stacktrace: message,
+                            });
```

Comment:
add this as an explicit error so it gets reported properly and the user can click `Reset` to start over

Reply:
Great! It's annoying to see unrecoverable errors.